### PR TITLE
Add React Native Support. for #10

### PR DESCRIPTION
**Pull request by gutenye**
A working proof-of-concept to enable React Native Support.

Tested on uploading a single file only, not working with multiple files.

### CODE REVIEW

**Comment by thebigredgeek** [comment]
Can we please get this merged in?

**Comment by jaydenseric** [comment]
@gutenye ❤️ Thanks for your work starting this off. This feature is in hot demand.

@thebigredgeek So far this is just a proof of concept and in it's broken state is a fair way off being ready. I will make some comments on the changes. Also, query batching is not accounted for.

We don't have build tests. I merged smaller updates than this that at first look appeared ok that inadvertently broke people's projects and took several patches to fix. I will be more conservative merging updates I am not 100% across myself.

I can't directly work on this enhancement as I am not setup for React Native. It would be great if someone with experience could step up and help to nurture this to completion.

File: package.json [hunk 1]
```diff
@@ -43,7 +43,8 @@
     "build:watch": "npm run build -- --watch",
     "prepublish": "npm run build"
   },
-  "main": "dist/apollo-upload-client.js",
+  "main": "src/index.js",
```
Comment (jaydenseric) [inline package.json]: What is the reason for this? We can't skip transpilation.

File: package.json [hunk 2]
```diff
@@ -43,7 +43,8 @@
...
+  "main1": "dist/apollo-upload-client.js",
```
Comment (jaydenseric) [inline package.json]: This doesn't do anything.

File: src/network-interface.js [hunk 3]
```diff
@@ -14,12 +17,12 @@ export class HTTPUploadNetworkInterface extends HTTPFetchNetworkInterface {
         operation.query = printAST(operation.query)
 
         // Build the form
-        const formData = new window.FormData()
+        const formData = new FormData()
```
Comment (jaydenseric) [inline src/network-interface.js]: Are there linting errors now that `window` is removed everywhere? ESLint config will need to be updated if this is a necessary change.

File: src/network-interface.js [hunk 4]
```diff
@@ -4,7 +4,10 @@ import {extractRequestFiles} from './helpers'
 export class HTTPUploadNetworkInterface extends HTTPFetchNetworkInterface {
   fetchFromRemoteEndpoint ({request, options}) {
     // Skip upload proccess if SSR
-    if (typeof window !== 'undefined') {
+    if (
+      (typeof window !== 'undefined') ||
+      (typeof navigator != 'undefined' && navigator.product == 'ReactNative')
```
Comment (jaydenseric) [inline src/network-interface.js]: `!=` is probably a linting error.

Perhaps we should just create one set of checks at the top of the module to define an agent const, and use it throughout the code to make it more DRY and readable.

Comment (jaydenseric) [inline src/network-interface.js]: Just realised that the checks appear in a few files, and in this one are not repeated.

File: package.json (hunk 1 above)
Comment (gutenye) [inline package.json]: My bad, I was testing it without transpilation, and forget to remove it after done.

File: package.json (hunk 2 above)
Comment (gutenye) [inline package.json]: fixed.

File: src/network-interface.js (hunk 3 above)
Comment (gutenye) [inline src/network-interface.js]: fixed by providing `standard.globals` in package.json

File: src/network-interface.js (hunk 4 above)
Comment (gutenye) [inline src/network-interface.js]: linting error is fixed.

**Comment by srtucker22** [comment]
Great work guys! Really looking forward to this PR

**Comment by jaydenseric** [comment]
@gutenye thanks for your work, the way forward can be seen [here](https://github.com/jaydenseric/apollo-upload-client/issues/10#issuecomment-310314433).

**Pull request closed without merging**